.travis.yml

```diff
@@ -1,36 +1,45 @@
-language: php
-
 os:
   - linux
 
+sudo: false
+
 git:
   depth: 1
 
-php:
-    - 7.1
-    - 7.2
-    - nightly
-
 matrix:
-  fast_finish: true
+  include:
+    - language: php
+      php: 7.1
+      before_install: .travis/php/before_install.sh
+      install: .travis/php/install.sh
+      before_script: .travis/php/before_script.sh
+      script: .travis/php/script.sh
```

Comment:
Why are we moving the script to external files? Cant we just use the list notation in yaml? For example:
```
script:
- bundle exec rake build
- bundle exec rake builddoc
```
Most of the scripts are short and having it all in one palce would make it easioer to maintain.

Reply:
I chose shell scripts over YAML because the same commands are run in several test environments (as for PHP it's 7.1, 7.2 and nightly versions).